crates/sui-adapter-transactional-tests/tests/txn_gas_api/gas_api.move

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//# init --simulator --accounts A --addresses test=0x0 --reference-gas-price 700 
+
+//# publish
+module test::gas_test;
+
+public fun gas_checks(price: u64, delta: u64, ctx: &TxContext) {
+    let rgp = ctx.reference_gas_price();
+    let gas_price = ctx.gas_price();
+    assert!(rgp == 700, 100);
+    assert!(gas_price == price , 101);
+    assert!(gas_price >= rgp + delta, 102);
```

Comment:
Would you mind duplicating this test for `test_scenario`? in `crates/sui-framework/packages/sui-framework/tests/test_scenario_tests.move` 

And do we need a way of setting these values in unit tests?

Reply:
added the test in test scenario, however had to change the way the context is built in test.
I have a test failing locally, though it's failing with or without this change so I suspect it may not be related to this PR and just a local failure, we'll see.
Not a big fan of how this is done now (see only last commit) and let me know